warp10/src/main/java/io/warp10/continuum/egress/EgressExecHandler.java

```diff
@@ -285,6 +303,66 @@ public void handle(String target, Request baseRequest, HttpServletRequest req, H
 
       boolean terminate = false;
 
+      //
+      // Read the X-Warp10-Timebox header
+      //
+
+      String timebox = req.getHeader(Constants.HTTP_HEADER_TIMEBOX);
+      long maxtime = 0L;
+
+      if (null != timebox) {
+        boolean fromToken = false;
+        try {
+          maxtime = Long.parseLong(timebox) * Constants.TIME_UNITS_PER_MS;
+        } catch (NumberFormatException nfe) {
+          try {
+            maxtime = DURATION.parseDuration(new Instant(), timebox, false, false);
+          } catch (Exception ee) {
+            resp.sendError(errorCode, "Invalid value for header " + Constants.HTTP_HEADER_TIMEBOX + " " + ThrowableUtils.getErrorMessage(ee));
+            return;
+          }
+        }
+
+        //
+        // Check the configured maxtime as the upper limit unless a valid token was specified as timebox
+        //
+
+        if (!fromToken && 0 != MAXTIME && maxtime > 0) {
+          maxtime = Math.min(maxtime, MAXTIME);
```

Comment:
If `maxtime < 0`, this will leave it negative thus removing the timebox, even if `MAXTIME > 0`.